cmake/CheckDGtalDependencies.cmake

```diff
@@ -20,6 +20,57 @@ if ( Boost_FOUND )
 endif( Boost_FOUND )
 
 # -----------------------------------------------------------------------------
+# Check some CPP11 features in the compiler
+# -----------------------------------------------------------------------------
+SET(C11_FOUND_DGTAL 0)
+SET(C11_AUTO_DGTAL 0)
+SET(C11_FORWARD_DGTAL 0)
+SET(C11_INITIALIZER_DGTAL 0)
+SET(C11_ARRAY 0)
+INCLUDE(CheckCPP11)
+IF (CPP11_INITIALIZER_LIST OR CPP11_AUTO OR CP11_FORWARD_LIST)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 ")
+  SET(C11_FOUND_DGTAL 1)
+  IF (CPP11_AUTO)
+    SET(C11_AUTO_DGTAL 1)
+    SET(C11_FEATURES "${C11_FEATURES} auto")
+  ENDIF()
+  IF (CPP11_INITIALIZER_LIST)
+    SET(C11_INITIALIZER_DGTAL 1)
+    SET(C11_FEATURES "${C11_FEATURES} initializer-list")
+  ENDIF()
+  IF (CPP11_FORWARD_LIST)
+    SET(C11_FORWARD_DGTAL 1)
+    SET(C11_FEATURES "${C11_FEATURES} std::forward-list")
+  ENDIF()
+  IF (CPP11_ARRAY)
+    SET(C11_ARRAY 1)
+    SET(C11_FEATURES "${C11_FEATURES} std::array")
+  ENDIF()
+  IF (CPP11_RREF_MOVE)
+    SET(C11_RREF_MOVE 1)
+    SET(C11_FEATURES "${C11_FEATURES} std::move rvalue-reference(&&)")
+  ENDIF()
+  MESSAGE(STATUS "Supported c++11 features: [${C11_FEATURES} ]")
+  ADD_DEFINITIONS("-DWITH_C11 ")
+ELSE()
+  MESSAGE(FATAL_ERROR "Your compiler does not support any c++11 feature. Please specify another C++ compiler of disable this WITH_C11 option.")
```

Comment:
since we can no more use it without c++11 we can update "or disable this WITH_C11 option"